### ISSUE TITLE
webdriver-manager start --detach do not detach the server on Windows 8

### ISSUE THREAD

**Issue by dnoliver** [angular/webdriver-manager#212]
using `--detach` option with start command do not put the server in background

output of `webdriver-manager start --detach`
```
$ webdriver-manager --detach start
[10:26:28] I/start - java -Dwebdriver.chrome.driver=C:\Users\dnoliver\AppData\Roaming\nvm\v6.7.0\node_modules\protractor\node_modules\webdriver-manager\selenium\chromedriver_2.27.exe -Dwebdriver.ie.driver=C:\Users\dnoliver\AppData\Roaming\nvm\v6.7.0\node_modules\protractor\node_modules\webdriver-manager\selenium\IEDriverServer3.1.0.exe -Dwebdriver.gecko.driver=C:\Users\dnoliver\AppData\Roaming\nvm\v6.7.0\node_modules\protractor\node_modules\webdriver-manager\selenium\geckodriver-v0.14.0.exe -jar C:\Users\dnoliver\AppData\Roaming\nvm\v6.7.0\node_modules\protractor\node_modules\webdriver-manager\selenium\selenium-server-standalone-3.1.0.jar -port 4444
[10:26:28] I/start - seleniumProcess.pid: 596
10:26:28.356 INFO - Selenium build info: version: '3.1.0', revision: '86a5d70'
10:26:28.357 INFO - Launching a standalone Selenium Server
2017-02-20 10:26:28.375:INFO::main: Logging initialized @293ms
10:26:28.454 INFO - Driver class not found: com.opera.core.systems.OperaDriver
10:26:28.454 INFO - Driver provider com.opera.core.systems.OperaDriver registration is skipped:
Unable to create new instances on this machine.
10:26:28.454 INFO - Driver class not found: com.opera.core.systems.OperaDriver
10:26:28.455 INFO - Driver provider com.opera.core.systems.OperaDriver is not registered
10:26:28.464 INFO - Driver provider org.openqa.selenium.safari.SafariDriver registration is skipped:
 registration capabilities Capabilities [{browserName=safari, version=, platform=MAC}] does not match the current platform WIN8_1
2017-02-20 10:26:28.517:INFO:osjs.Server:main: jetty-9.2.15.v20160210
2017-02-20 10:26:28.556:INFO:osjsh.ContextHandler:main: Started o.s.j.s.ServletContextHandler@4520ebad{/,null,AVAILABLE}
2017-02-20 10:26:29.072:INFO:osjs.ServerConnector:main: Started ServerConnector@4497bf4c{HTTP/1.1}{0.0.0.0:4444}
2017-02-20 10:26:29.073:INFO:osjs.Server:main: Started @991ms
10:26:29.073 INFO - Selenium Server is up and running
```

System info:

OS: Windows 8
Node Version: v6.7.0
NPM version: v4.2.0
java version "1.8.0_102"
Java(TM) SE Runtime Environment (build 1.8.0_102-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.102-b14, mixed mode)
Picked up _JAVA_OPTIONS: -Djava.net.preferIPv4Stack=true
webdriver-manager 12.0.2
Working in a network with proxy configuration

**Comment by Sleepyowl** [comment]
Confirmed with Windows 10 (10.0.14393) too

**Comment by dnoliver** [comment]
Also happened to me in a Docker container with Alpine Linux. But in https://hub.docker.com/r/caltha/protractor/ image, --detach option works well.

**Comment by duffman03** [comment]
I reproduced this issue as well. Any workarounds to start/stop the process in one test script?

OS:
ProductName:	Mac OS X
ProductVersion:	10.12.3
BuildVersion:	16D32
Node:
v6.9.4
Webdriver-manager:
12.0.2

**Comment by elhigu** [comment]
@duffman03 my workaround was to downgrade to version webdriver-manager@11.1.1

**Comment by wiktormarchewka** [comment]
**OS:** macOS Sierra 10.12.3
**webdriver-manager:** 12.0.3
I reporoduce this as well as other people.

**Comment by lukaserat** [comment]
Same happens on my setup

OS: macOS Sierra 10.12.4
node: v6.9.4
npm: 4.1.2
webdriver-manager: 12.0.4

**Comment by atd** [comment]
Same here

OS: Debian GNU/Linux testing
node: v6.10.2
npm: 3.10.10
webdriver-manager: 12.0.5

**Comment by lgwillmore** [comment]
I am sure it is been worked on - just wanted to keep it alive. My stuff:
GNU/Linux
Node: v7.10.0
webdriver-manager: 12.0.5

**Comment by inad9300** [comment]
Same here in Windows 10 and Node 6.10.0. Any plans on dealing with this issue?

**Comment by ayaz-ashrapov** [comment]
Same here
OS: Windows Server 2012 R2
node: v6.11.0
npm: 3.10.10
webdriver-manager: 12.0.6

**Comment by shcallaway** [comment]
Bump.

OS: macOS Sierra 10.12.4
node: 6.10.2
npm: 3.10.10
webdriver-manager: 12.0.6

**Comment by Remimstr** [comment]
Also having this problem
OS: macOS Sierra 10.12.4
node: 6.10.2
npm: 3.10.10
webdriver-manager: 12.0.6

**Comment by JimSangwine** [comment]
Me too.

OS: Ubuntu Gnome 17.04
node: 6.9.1
npm: 3.10.8

**Comment by bengerman13** [comment]
I'm having this issue as well in a Fedora 25 docker container. I noticed this error after letting it sit for a while:  
     ` [root@9bd947e1fd2b e2e]# [19:48:35] E/start - Timed out wating forselenium server.  Final rejection reason: {"code":"ECONNREFUSED","errno":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":80}`

So I tried this and it worked:  
`webdriver-manager start --detach --seleniumPort=80`

**Comment by bridzius** [comment]
Ran into same issue using newest protractor bundled `webdriver-manager`

OS: macOS Sierra 10.12.6
NodeJS: 7.4.0
npm: 5.4.0
yarn: 0.27.5
webdriver-manager: 12.0.6

**Comment by bdellegrazie** [comment]
I can confirm that @bengerman13's fix does indeed fix this issue.

**Comment by elhigu** [comment]
@bengerman13 @bdellegrazie I can confirm that `webdriver-manager start --detach --seleniumPort=80` is not a solution for this issue (--detach not working with newer selenium). The @bengerman13's problem is completely different issue.

**Comment by bdellegrazie** [comment]
@elhigu please check the *fix* provided to PR #268 and #266, not the comments.

The code appears to wait on the appium port instead of the selenium port. All the PR does is wait on the selenium port instead.

The symptoms you, others and I have observed likely stem from the same issue - the code is waiting on a port that is never opened because we don't have Android SDK installed or enabled on the command line so the process never detaches.

w.r.t. specifying "--seleniumPort=80" I agree - that won't help at all.

**Comment by elhigu** [comment]
@bdellegrazie I tried that patch too and it really did seem to start as detached afterwards and looking from the code that looked like a bug. Continued discussion about the PR in its comments.

Also when detach starts working, shutdown will still break anyways after >= 12 because of https://github.com/angular/webdriver-manager/issues/199

**Comment by sahbi-ktifa** [comment]
It may only be a quick workaround, but running : 
`webdriver-manager start --detach --seleniumPort=80 &`
seems to do the trick as server runs as background task.

It is not a fix although but a quick and temporary workaround.

**Comment by elhigu** [comment]
@sahbi-ktifa you still cannot shut it down if started like that + there is no need to add any extra cmd options if you are using `&` for starting program to background...

**Comment by thiagolunardi** [comment]
Same there:

Windows 10 build 1706
NodeJS: 6.10.0
Java: 9.0.1
webdriver-manager: 12.0.6

![image](https://user-images.githubusercontent.com/5428523/31760253-5f97313a-b4b4-11e7-88c5-17fb64dcce00.png)


@dnoliver I suggest to change the issue title, as we are experiencing in several OS flavors.

**Comment by msitnikov** [comment]
Any idea on the estimated time of arrival on this fix?

**Comment by thiagolunardi** [comment]
After some update, still the same:

Windows 10 Enterprise Version 1709 build 16299.19
NodeJS: 8.9.1 LTS
Java: 9.0.1
webdriver-manager: 12.0.6

**Comment by elhigu** [comment]
Since fix for this doesn't seem to be merged (`--detach` and `stop` commands) and our tests doesn't work anymore with older drivers and newer chrome, I had to came up with new workaround for this.

Now I use `pm2` to  manage webdriver manager like this: 

```
# start webdriver
pm2 start webdriver-manager --name=webdriver -- start

# stop webdriver
pm2 delete webdriver
```

Nice thing with this workaround is that it also fixes the problem, that webdriver manager hangs due to stream buffer overflow or something like that if quiet mode is used and selenium outputs debug messages too fast rate (like  when polling stuff in javascript side).

**Comment by gvsrepins** [comment]
Hi guys, any update on this issue?

**Comment by felipecardona** [comment]
Hey guys, any clue on how to implement this, ill offer myself to implement it and ask for PR :D, a friend of mine is suffering :D

**Comment by elhigu** [comment]
@felipecardona this has already been implemented and PRs had been sent months ago, but maintainers just haven’t merged them.

**Comment by elhigu** [comment]
Looks like #266 and #268 were already closed by #228. PR fixing shutting down the server afterwards is #274.

Anyways using pm2 to handle starting/stopping process has worked fine for me lately.

**Comment by ErikSchierboom** [comment]
> Is webdriver-manager still maintained?

I'm also interested in this. To me, it looks like this issue might have been fixed when #228 was merged, however I cannot test that as it was merged after the latest release, which was on *May 9, 2017*! That is over a year ago! Is there still someone that maintains this project and can build releases?

**Comment by cnishina** [comment]
I was hoping the new maintainer of Protractor would have cut a release on this too. I'll cut this release some time next week. Since its been a while, should we do a major or minor version release?

**Comment by ErikSchierboom** [comment]
Ehm, I don't exactly know how many changes were made after the previous release, but if it isn't anything major, I would probably do a minor release.

**Comment by cnishina** [comment]
I have it ready to go and will let the maintainer publish it. The new release should be 12.1.0

**Comment by ErikSchierboom** [comment]
Great!

**Comment by toddobryan-clever** [comment]
I'm still seeing this problem with webdriver-manager 12.1.0 on Mac OS 10.13.6

**Comment by cnishina** [comment]
So in the master branch this has been fixed. I've tested this on Windows, Mac, and Linux. https://github.com/angular/webdriver-manager/blob/master/lib/provider/selenium_server.ts#L122

Closing this issue as fixed. This is sort of released in version 13 although it will not work on Protractor 5.4.2 (this has to do with driver providers: direct connect and local). I am still working on fixes to Protractor 6.
